Feat: Authentication

bharati-21 opened the pull request:
This PR contains the following
- Users can login/ signup to the app
- Without logging in, they cannot access private routes
- Slice for auth state and reducers to update auth state
- Login, signup services
- Logout

bharati-21 committed:
feat: add screen breakpoints

bharati-21 committed:
feat: navabr. footer components

bharati-21 committed:
chore: installations, config files

bharati-21 committed:
feat: font awesome icon kit

bharati-21 committed:
feat: render navbar, footer, landing page

bharati-21 committed:
feat: supporting image assets

bharati-21 committed:
styles: set google fonts

bharati-21 committed:
feat: landing page

bharati-21 committed:
class styles: remove underline

bharati-21 committed:
feat: added login button

bharati-21 committed:
feat: renamed pages to features

bharati-21 committed:
chore: installation

bharati-21 committed:
chore: installations

bharati-21 committed:
feat: toast container portal

bharati-21 committed:
feat: added navroutes

bharati-21 committed:
feat: auth slice with reducers and actions

bharati-21 committed:
feat: auth login, signup pages

bharati-21 committed:
feat: barrel export

bharati-21 committed:
feat: custom hook to display toast

bharati-21 committed:
feat: custom hook to change title

bharati-21 committed:
fix: edit title from login to signup

bharati-21 committed:
feat: logout feature

bharati-21 committed:
feat: hooks barrel export

bharati-21 committed:
feat: configure store with reducers

bharati-21 committed:
feat: portal for toast container

bharati-21 committed:
feat: login, logout service

bharati-21 committed:
feat: routes

bharati-21 committed:
styles: components, toastify

bharati-21 committed:
export: services barrel export

bharati-21 committed:
feat: changed button to link

bharati-21 committed:
feat: wrapped provider and store. added portal

bharati-21 committed:
feat: starter template for Home Page

bharati-21 committed:
Merge branch 'development' of https://github.com/bharati-21/readers-space into auth

jeetbhalani15 reviewed (approved):
Auth-services works fine
- Code looks readable
- nice UI

sonishreyas reviewed (approved):
The code looks good. Got to learn about portals from your code, will try to implement it in my code. The UI also looks good.

pruthvirajmv commented on src/features/Auth/Login/index.jsx:
```diff
@@ -0,0 +1,144 @@
+	const handleLoginWithTestCredentials = () => {
+		setFormData({
+			username: "adarshbalika",
+			password: "adarshBalika123",
```
please keep keys/creds in .env

bharati-21 replied:
Yes, I have made these changes.

pruthvirajmv commented on src/features/Auth/Login/index.jsx:
```diff
@@ -0,0 +1,144 @@
+							htmlFor="input-username"
+							className="auth-form-label"
+						>
+							Username <span className="text-red-500">*</span>
```
just try how can you make this star using css

bharati-21 replied:
Yes, I used the ::after pseudo element and created the star

pruthvirajmv commented on src/features/Auth/Login/index.jsx:
```diff
@@ -0,0 +1,144 @@
+								<input
+									type={`${
+										showPassword ? "text" : "password"
+									}`}
```
please check this would work herer {showPassword ? "text" : "password"}

bharati-21 replied:
Yes, I have made the changes

pruthvirajmv commented on src/features/Auth/Signup/index.jsx:
```diff
@@ -0,0 +1,236 @@
+			}
+			throw new Error("Signup Failed");
+		} catch (error) {
+			showToast("Signup Failed.", "error");
```
please check this logic

bharati-21 replied:
Yes, I have fixed this issue.

bharati-21 replied:
> Hey Aditi,
> 
> Good to see redux-toolkit implementation Please go through the comments as discussed & resolve
> 
> Keep coding!

Thank you for your time and review! I have implemented the suggested changes.
To ensure the logo image & text in the navbar is a link - have implemented this in the next PR.
Redirect to home page when authenticated user tries to land in landing page - I have implemented this as well in the next PR.

Thank you once again for your review.

bharati-21 replied:
> Hey Aditi,
> 
> Good to see redux-toolkit implementation Please go through the comments as discussed & resolve
> 
> Keep coding!

Thank you for your time and review! I have implemented the suggested changes.
To ensure the logo image & text in the navbar is a link - have implemented this in the next PR.
Redirect to home page when authenticated user tries to land in landing page - I have implemented this as well in the next PR.

Thank you once again for your review.

bharati-21 replied:
> Auth-services works fine
> 
> * Code looks readable
> * nice UI

Thank you for your review!

pruthvirajmv commented on src/features/Auth/Signup/index.jsx:
```diff
@@ -0,0 +1,236 @@
+							<span className="text-red-500">*</span>
+							<div className="input-container w-full relative">
+								<input
+									type={`${
```
check the logic

bharati-21 replied:
Yes, I have fixed this issue.

pruthvirajmv reviewed (commented):
Hey Aditi,

Good to see redux-toolkit implementation
Please go through the comments as discussed & resolve

Keep coding!

bharati-21 committed:
feat: use env variables for guest login

bharati-21 committed:
styles: add required character using css

bharati-21 committed:
Update index.jsx

bharati-21 committed:
fix: login, logout messages

bharati-21 committed:
feat: adding logs to check error

bharati-21 commented:
> The code looks good. Got to learn about portals from your code, will try to implement it in my code. The UI also looks good.

Thank you for your review!

bharati-21 commented:
> Hey Aditi,
> 
> Good to see redux-toolkit implementation Please go through the comments as discussed & resolve
> 
> Keep coding!

Thank you for your time and review! I have implemented the suggested changes.
To ensure the logo image & text in the navbar is a link - have implemented this in the next PR.
Redirect to home page when authenticated user tries to land in landing page - I have implemented this as well in the next PR.

Thank you once again for your review.
